### DISCUSSION TITLE
table index is nil python ruff

### DISCUSSION THREAD

**Discussion by DavidEdsonDoNascimento** [LazyVim/LazyVim#4206] [Q&A] [answered]
I'm new to vim and I'm using lazyvim, I configured the lsp pyright and ruff according to the lazyvim documentation as shown in: 
https://www.lazyvim.org/extras/lang/python 
but I get the error: table index is nil 
exactly on the ruff configuration line:

![image](https://github.com/user-attachments/assets/b01c990b-444c-46f4-8e93-f313fa5039e1)

error returned:

![image](https://github.com/user-attachments/assets/0a8d5155-fe8f-488c-9d31-b1caa55df50f)

I've been at this for a few days, if anyone can help me I'd be grateful, I've searched the internet and haven't found anything that makes sense

Mason:
![image](https://github.com/user-attachments/assets/0619af05-ccfc-40fa-8fa1-c05d6e9377ed)

my dotfiles:
https://github.com/DavidEdsonDoNascimento/dotfiles

**Comment by dpetka2001** [comment, marked as answer]
You don't have to copy over everything that is mentioned on the website. You just have to enable the Python Extra as is mentioned at the top of the page. Everything mentioned on the website is supposed to be taken as a reference for what the default Lazyvim configuration is like.

**Reply by DavidEdsonDoNascimento** [reply]
Sorry, I started using Vim a few days ago. I thought that for it to work completely I would need to install and configure Pyright and Ruff, but it's true after I removed the lsp settings it worked, I believe that just with what's in Python Extra it should be enough, thanks
